src/components/ProjectIconSelection/ProjectIconSelection.js

```diff
@@ -0,0 +1,86 @@
+// @flow
+import React, { Component } from 'react';
+import styled from 'styled-components';
+import importAll from 'import-all.macro';
+
+import { sampleMany } from '../../utils';
+import { BREAKPOINTS } from '../../constants';
+
+import SelectableImage from '../SelectableImage';
+
+const icons: Array<mixed> = importAll.sync(
+  '../../assets/images/icons/icon_*.*'
+);
+const iconSrcs: Array<string> = Object.values(icons).map(src => String(src));
+
+const DEFAULT_SUBSET = 10;
+const DEFAULT_ICON_SIZE = 60;
+
+type Props = {
+  selectedIcon: ?string,
+  showRandomSubset: boolean | number,
```

Comment:
Hm, so the more I think about it, the more I think we probably want this to be two props: `limitTo: number` and `randomize: boolean`.

Right now, the ProjectSettingsModal includes _all_ icons, but that won't be the case forever as we add more icons. So I think we'll always want to limit it to a certain number (for now, we can just set that number to 21, since that's the number we actually have).

And then `randomize` can let us know if we just want to take the first N, or take a random N.

Sorry for not thinking of this sooner!